### PR TITLE
Add DeathAdder V3 HyperSpeed

### LINKED ISSUE #2294: Support for Razer DeathAdder V3 HyperSpeed
### Device Name

Razer DeathAdder V3 HyperSpeed

### Product Number

RZ01-05140100

### Model Number

RZ01-0514

### VID:PID

1532:00c5

### USB Details

```shell
Bus 001 Device 003: ID 1532:00c5 Razer USA, Ltd Razer DeathAdder V3 HyperSpeed
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 [unknown]
  bDeviceSubClass         0 [unknown]
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x00c5 Razer DeathAdder V3 HyperSpeed
  bcdDevice            1.00
  iManufacturer           1 Razer
  iProduct                2 Razer DeathAdder V3 HyperSpeed
  iSerial                 3 000000000000
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          4 Default configuration
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              100mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
          Report Descriptors: 
            ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     314
          Report Descriptors: 
            ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
          Report Descriptors: 
            ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0040  1x 64 bytes
        bInterval               1
Device Status:     0x0000
  (Bus Powered)
```

### Device Input Paths

```shell
drwxr-xr-x 2 root root 300 Aug 30 16:53 .
drwxr-xr-x 4 root root 700 Aug 30 16:53 ..
lrwxrwxrwx 1 root root  10 Aug 30 16:53 usb-Razer_Razer_DeathAdder_V3_HyperSpeed_000000000000-event-if01 -> ../event17
lrwxrwxrwx 1 root root  10 Aug 30 16:53 usb-Razer_Razer_DeathAdder_V3_HyperSpeed_000000000000-event-mouse -> ../event14
lrwxrwxrwx 1 root root  10 Aug 30 16:53 usb-Razer_Razer_DeathAdder_V3_HyperSpeed_000000000000-if01-event-kbd -> ../event15
lrwxrwxrwx 1 root root  10 Aug 30 16:53 usb-Razer_Razer_DeathAdder_V3_HyperSpeed_000000000000-if01-event-mouse -> ../event16
lrwxrwxrwx 1 root root   9 Aug 30 16:53 usb-Razer_Razer_DeathAdder_V3_HyperSpeed_000000000000-if01-mouse -> ../mouse1
lrwxrwxrwx 1 root root  10 Aug 30 16:53 usb-Razer_Razer_DeathAdder_V3_HyperSpeed_000000000000-if02-event-kbd -> ../event18
lrwxrwxrwx 1 root root   9 Aug 30 16:53 usb-Razer_Razer_DeathAdder_V3_HyperSpeed_000000000000-mouse -> ../mouse0
```

### USB HID Dump

```shell
001:003:002:DESCRIPTOR         1725062438.993413
 05 01 09 06 A1 01 05 07 19 E0 29 E7 15 00 25 01
 75 01 95 08 81 02 81 01 19 00 2A FF 00 15 00 26
 FF 00 75 08 95 06 81 00 05 08 19 01 29 03 15 00
 25 01 75 01 95 03 91 02 95 05 91 01 C0

001:003:001:DESCRIPTOR         1725062439.000415
 05 01 09 06 A1 01 85 01 05 07 19 E0 29 E7 15 00
 25 01 75 01 95 08 81 02 19 00 2A FF 00 15 00 26
 FF 00 75 08 95 0E 81 00 05 08 19 01 29 03 15 00
 25 01 75 01 95 03 91 02 95 05 91 01 C0 05 01 09
 02 A1 01 85 0A 09 01 A1 00 05 09 19 01 29 06 15
 00 25 01 75 01 95 06 81 02 75 01 95 02 81 03 06
 00 FF 09 40 75 08 95 02 15 81 25 7F 81 02 05 01
 09 38 15 81 25 7F 75 08 95 01 81 06 09 30 09 31
 16 00 80 26 FF 7F 75 10 95 02 81 06 C0 C0 05 0C
 09 01 A1 01 85 02 19 00 2A 3C 02 15 00 26 3C 02
 95 01 75 10 81 00 75 08 95 0D 81 01 C0 05 01 09
 80 A1 01 85 03 19 81 29 83 15 00 25 01 75 01 95
 03 81 02 95 05 81 01 75 08 95 0E 81 01 C0 05 01
 09 00 A1 01 85 04 09 03 15 00 26 FF 00 35 00 46
 FF 00 75 08 95 0F 81 00 C0 05 01 09 00 A1 01 85
 05 09 03 15 00 26 FF 00 35 00 46 FF 00 75 08 95
 0F 81 00 C0 05 01 09 00 A1 01 85 08 09 03 15 00
 26 FF 00 35 00 46 FF 00 75 08 95 0F 81 00 C0 05
 01 09 00 A1 01 85 09 09 03 15 00 26 FF 00 35 00
 46 FF 00 75 08 95 0F 81 00 C0

001:003:000:DESCRIPTOR         1725062439.004413
 05 01 09 02 A1 01 09 01 A1 00 05 09 19 01 29 05
 15 00 25 01 75 01 95 05 81 02 75 01 95 0B 81 03
 05 0C 0A 38 02 15 81 25 7F 75 08 95 01 81 06 05
 01 09 38 15 81 25 7F 75 08 95 01 81 06 09 30 09
 31 16 01 80 26 FF 7F 75 10 95 02 81 06 C0 06 00
 FF 09 02 15 00 25 01 75 08 95 5A B1 01 C0
```

### Packet Captures

_No response_

### Synapse Screenshots

Screenshot 2024-08-30 161752
Screenshot 2024-08-30 161802
Screenshot 2024-08-30 161812
Screenshot 2024-08-30 161817
Screenshot 2024-08-30 161828

### PR DESCRIPTION
**Pull request by stuartf**
fixes #2294

### CODE REVIEW

File: daemon/openrazer_daemon/hardware/mouse.py [hunk 1]
```diff
@@ -1841,3 +1841,31 @@ class RazerViperV3ProWireless(RazerViperV3ProWired):
...
+
+
+class RazerDeathAdderV3HyperSpeedWireless(RazerDeathAdderV3HyperSpeedWired):
+    """
+    Class for the Razer DeathAdder V3 HyperSpeed (Wireless)
+    """
+
+    USB_PID = 0x00C5
+
+    POLL_RATES = [125, 500, 1000]
```
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/mouse.py]: This is the default, why are you setting it explicitly for the Wireless variant? Or are you missing to set it in the Wired variant?

**Comment by z3ntu** [comment]
The rest of the diff looks sane, thanks!

I posted a call for testing in https://github.com/openrazer/openrazer/issues/2294#issuecomment-2424799217, let's see if someone else can test this before merging.

File: daemon/openrazer_daemon/hardware/mouse.py (hunk 1 above)
Comment (stuartf) [reply daemon/openrazer_daemon/hardware/mouse.py]: I didn't set it in the wired variant, wasn't sure if I needed to set it for the wireless.

Comment (z3ntu) [reply daemon/openrazer_daemon/hardware/mouse.py]: `POLL_RATES = [125, 500, 1000]` is default, so you can just remove it if this device doesn't have poll rate > 1000Hz support or some other weird limitation.

**Comment by z3ntu** [comment]
We're having some CI troubles, once https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/76653 is merged we should be good again.

edit: and https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/76662

**Review by z3ntu — APPROVED** [review]
LGTM, thank you!

**Pull request merged**